internal/store/store.go

```diff
@@ -324,7 +330,7 @@ func (bs *BlockStore) PruneBlocks(height int64, state sm.State) (uint64, int64,
 		bs.mtx.Lock()
 		bs.base = base
 		bs.mtx.Unlock()
-		bs.saveState()
+		bs.saveState(batch)
```

Comment:
Could we rename this variable so it does not shadow the outer one?

Reply:
Not related to this PR, but this function closes the batch, but this function is called on a batch for which `defer Close()` has also been called. The second close should be a nop, but it would be better to clearly specify who closes it.

Reply:
If the close inside the function remains, then it should be a deferred call.